src/data_gradients/dataset_adapters/formatters/detection.py

```diff
@@ -40,6 +41,11 @@ def format(self, images: Tensor, labels: Tensor) -> Tuple[Tensor, Tensor]:
             - labels: List of bounding boxes, each of shape (N_i, 5 [label_xyxy]) with N_i being the number of bounding boxes with class_id in class_ids
         """
 
+        if self.class_ids_to_use is None:
+            # This may trigger questions to the user, so we prefer to set it inside `former()` and not `__init__`
+            # to avoid asking questions even before the analysis starts.
+            self.class_ids_to_use = [self.data_config.get_class_names().index(class_name) for class_name in self.data_config.get_class_names_to_use()]
```

Comment:
if class names were not provided, and the user answered the prompt with a number, with this for loop return "c1, c2, .."?

Reply:
Yeah something like that. Maybe ["class_1", ...], I dont recall exactlyt the syntax.